### PR TITLE
Add auto-approve capabilities. Closes #16.

### LINKED ISSUE #16: Add an auto approval follow for classified data
In the natural language flow the user is currently prompted to select the appropriate command identifier if their phrase was of medium confidence. All classified data is currently placed into a local database (Pouch) for training data purposes (high, medium, low). In the existing flow the user must approve the untrained data to be included in the next re-training cycle by approving the recorded training data. 

However, if the user wishes for their training data to be automatically approved additional functionality is required. A `HUBOT_NLC_AUTO_APPROVE=true` environment variable will be added to enable this flow. Any data record with a selected class provided will be automatically approved if the above variable is set to true.

### PR DESCRIPTION
**Pull request by chambrid**

### CODE REVIEW

**Comment by jlpadilla** [comment]
To keep the code consistent, could we use `env.js` to load the environment variable?  I'd move the `thruthy` function to `env.js` too as it could be shared by other envs in the future.

File: src/lib/nlcDb.js [hunk 1]
```diff
@@ -212,8 +222,12 @@ module.exports.post = function(classification, type, selectedClass) {
 			doc.classification = classification;
 		}
 
-		if (selectedClass)
+		if (selectedClass) {
 			doc.selectedClass = selectedClass;
+			if (truthy(process.env.HUBOT_WATSON_NLC_AUTO_APPROVE)) {
+				doc.approved = true;
```
Comment (jlpadilla) [inline src/lib/nlcDb.js:218]: Need to confirm with @normanb , but I believe the UI sets `doc.approved` to a timestamp indicating when the document was approved.

Comment (jlpadilla) [inline src/lib/nlcDb.js:218]: Confirmed that the UI is setting `doc.approved: Date.now()`

``` js
if(this.parent.type == 'approved') {
                let changedData = {
                    'id': ev.item.doc.id,
                    'text': this.statementTextApproved.value,
                    'selectedClass': this.classTextApproved.value,
                    'approved': Date.now()
                }
                util.putDBData('/api/favorites', changedData).then(function(res) {
                    self.update();
                })
            }
```

**Comment by chambrid** [comment]
@jlpadilla Updates made.

**Comment by jlpadilla** [comment]
From @normanb : 

> Just saw the last commit using `approved` as a timestamp, it works and doesn’t break the pouchdb design function, but it isn’t very contextual. I think `approved` as a boolean and then a timestamp would be more intuitive. I see it is in the UI, but it might be something we want to change. 

I like Norman's recommendation of separating it in 2 fields. Also, I'd like to add a flag `approved_method= 'auto'`. Having this data would make it easy to identify, analyze, and if needed remove these entries from the training data at any time in the future. I'm thinking we could have multiple approve methods like `auto, manual, bulk, or advanced-auto-approve based on analytics`. So I'm thinking we should add something like the following:

``` js
if (env.truthy(env.nlc_autoApprove)) {
    doc.approved = true;
    doc.approved_timestamp = Date.now();
    doc.approved_method = 'auto';
 }
```

**Comment by chambrid** [comment]
@jlpadilla update requested made

**Comment by chambrid** [comment]
@jlpadilla Added more content to the test case, so it matches the approval information discussed above.

**Comment by jlpadilla** [comment]
+1

Thanks for the rework!

**Pull request merged**